grails-app/services/I2b2HelperService.groovy

```diff
@@ -755,15 +755,24 @@ class I2b2HelperService {
     def ExportTableNew addConceptDataToTable(ExportTableNew tablein, String concept_key, String result_instance_id) {
         checkQueryResultAccess result_instance_id
 
+        /* As the column headers only show the (in many cases ambiguous) leaf part of the concept path,
+         * showing the full concept path in the tooltip is much more informative.
+         * As no tooltip text is passed on to the GridView code, the value of the string columnid is used
+         * and shown as the tooltip text when hoovering over the column header in GridView.
+         * Explicitly passing a tooltip text to the GridView code removes the necessity to use this columnid value.
+         * Removal of some undesired non-alpha-numeric characters from tooltip string
+         * prevents display errors in GridView (drop down menu, columns not showing or cells not being filled).
+         */
         String columnid = concept_key.encodeAsSHA1()
         String columnname = getColumnNameFromKey(concept_key).replace(" ", "_")
+        String columntooltip = keyToPath(concept_key).replace(" ", "_").replace(".","_").replace("[","_")
```

Comment:
I think it'd be hard to come up with "black list" of symbols. Wouldn't it be easier to replace all non non-word characters? Something like this: `keyToPath(concept_key).replaceAll('[^a-zA-Z_0-9\\\\]+', '_')`. Note we do not replace backslash here `\\\\`.